### ISSUE TITLE
why so many noise?

### ISSUE THREAD

**Issue by AIHHU** [etzinis/sudo_rm_rf#12]
I tried improved_rm_rf on wham clean speech separation and the separation result had so many noise, And its Si-SDR is 15.0

**Comment by etzinis** [comment]
I don't know which version you tried, I am sure you are doing something wrong though (maybe you do not normalize the waveform?). As specified in the README, you can see how to use the pre-trained models here: https://github.com/etzinis/sudo_rm_rf/blob/master/sudo_rm_rf/notebooks/sudormrf_how_to_use.ipynb

**Comment by etzinis** [comment]
Moreover, SI-SDR at 15 dB corresponds no noise at all, so please check again your code.

**Comment by AIHHU** [comment]
Thank you sir, I tried WHAM clean separation experiment as in your readme doc and did the separation as in your notebook. Normalization is also tried.

**Comment by etzinis** [comment]
Maybe you are getting the wrong file or something, in any case, it's almost impossible to get a really noisy file with 15dB SI-SDR. Except of the case where the noisy file is your reference signal :P

**Comment by AIHHU** [comment]
The only difference is that i use sf.read instead of torchaudio.read and use keep_dims intead of keep_dim

**Comment by etzinis** [comment]
It's impossible to debug your code without looking at it. Please use what I have in my notebook and I am sure it will run smoothly.

**Comment by AIHHU** [comment]
Thanks. i'll try it soon. I can ensure that i don't change your experiment code and the dataset is the wham dataset. That's true.

**Comment by AIHHU** [comment]
That's my inference code sir. It's all same to you. Why there also exits so many noise. The returned test Si-SDR index is 15.0.  The model is trained on 16khz,.
      esti_utt, _ = torchaudio.load(os.path.join(mix_file_path, file_id))
            
            input_mix_std = esti_utt.std(-1, keepdim=True)
            input_mix_mean = esti_utt.mean(-1, keepdim=True)
            input_mix = (esti_utt - input_mix_mean) / (input_mix_std + 1e-9)

            rec_sources_wavs = model_separation(input_mix.unsqueeze(1))

    
            #rec_sources_wavs = (rec_sources_wavs * input_mix_std) + input_mix_mean
        
            est_waveform_1 = rec_sources_wavs[0, 0].detach().numpy()
            est_waveform_2 = rec_sources_wavs[0, 1].detach().numpy()
            
            #print(rec_sources_wavs.shape)
            
            sf.write(os.path.join(esti_file_path, '1'+file_id), est_waveform_1, args.fs)
            sf.write(os.path.join(esti_file_path, '2'+file_id), est_waveform_2, args.fs)

**Comment by AIHHU** [comment]
The # in the line 
#rec_sources_wavs = (rec_sources_wavs * input_mix_std) + input_mix_mean is removed when being tested

**Comment by etzinis** [comment]
You are still giving me incomplete code snippets - tbh I don't see the reason for me to debug your code if you have a notebook which is bug-free. Are you pointing to files which are 8kHz ? Rescaling is important for capturing the appropriate gain of the sources.

**Comment by AIHHU** [comment]
nope, sir, I have a question: if i use the 16khz dataset to train and change the --fs in the command line. Will it train on the 16khz dataset? or In your code, it will resample to 8khz?

**Comment by etzinis** [comment]
those are 8khz models, you have to downsample your files to 8kHz first and then process them.

**Comment by AIHHU** [comment]
sorry to hear that, lol, 16khz model is required

**Comment by etzinis** [comment]
You mean is required for your application? I was thinking that maybe it would also be a good idea to run a couple of 16 kHz models.
